### DISCUSSION TITLE
Importing notebook #!import

### DISCUSSION THREAD

**Discussion by rajyraman** [dotnet/interactive#2426] [Q&A] [answered]
I was trying out #!import other day and it works fine.

![image](https://user-images.githubusercontent.com/5035266/199845446-521eb31d-66e5-4151-9115-a79115497c4e.png)

In the above screenshot you can see that I am using RunSQL function that is defined in the "SQL 4 CDS Main" notebook. I find this a bit clunky. Is there are anyway to generate a magic command on the fly when the parent notebook is referenced and executed on the child notebook, or offload the cell execution to the parent notebook for specific types of cells?

**Comment by jonsequitur** [comment, marked as answer]
You can add magic commands or subkernels on the fly.

<img width="657" alt="image" src="https://user-images.githubusercontent.com/547415/199854909-263ae197-3f11-442c-b914-d7536f1fbc4a.png">

For more information on how to add options and so on to the magic command, have a look at the System.CommandLine documentation: https://learn.microsoft.com/en-us/dotnet/standard/commandline/

**Reply by rajyraman** [reply]
@jonsequitur - That's awesome. I will give it a try. Thank you John.
